experimental/sql/test/ibis_dialect_tests/simple_query.xdsl

```diff
@@ -0,0 +1,52 @@
+// RUN: rel_opt.py %s | filecheck %s
+
+// Query: table.filter(table['a'] == 'AS')
+module() {
```

Comment:
Thanks for splitting this up !

Upon discussion with Ingo, it appears that it is not very clear to me what purpose the SSA values serve in the ibis dialect.

I'd suggest making everything a terminator to encode a tree.

Later we can rewrite the tree to SSA form by looking at symbols (e.g. table name) which I understand carry the reuse semantics ?

If we were to only use terminators everywhere, the IR would look like:

```suggestion
module() {
  ibis.selection() {
    // selection input
    ibis.pandas_table() ["table_name" = "t"] {
      ibis.schema_element() ["elt_name" = "a", "elt_type" = !ibis.String<1 : !i1>]
    }
  } {
    // selection predicate
    ibis.equals() {
      // predicate column
      ibis.table_column() ["col_name" = "a"] {
        ibis.pandas_table() ["table_name" = "t"] {
          ibis.schema_element() ["elt_name" = "a", "elt_type" = !ibis.String<1 : !i1>]
        }
      }
    } {
      // predicate literal
      ibis.literal() ["val" = "AS", "type" = !ibis.String<1 : !i1>]
    }
  }
}
```

What do you think of this modeling ?